File: src/main/java/org/hisp/dhis/rules/models/RuleActionAssign.java
package org.hisp.dhis.rules.models;

import javax.annotation.CheckForNull;
import javax.annotation.Nonnull;

import static org.hisp.dhis.rules.models.AttributeType.UNKNOWN;

;

public record RuleActionAssign(
        @Nonnull String data,
        @Nonnull AttributeType attributeType,
        @Nonnull String content,
        @Nonnull String field
) implements RuleActionAttribute {

    @Nonnull
    public static RuleActionAssign create(@CheckForNull String content,
                                          @Nonnull String data, @CheckForNull String field, @CheckForNull AttributeType attributeType) {
        if (content == null && field == null) {
            throw new IllegalArgumentException("Either content or field " +
                    "parameters must be not null.");
        }
        return new RuleActionAssign(data, attributeType == null ? UNKNOWN : attributeType, content == null ? "" : content,
                field == null ? "" : field);
    }

    @Nonnull
    public static RuleActionAssign create(@CheckForNull String content, @Nonnull String data, @CheckForNull String field) {
        return create(content, data, field, UNKNOWN);
    }
}

File: src/main/java/org/hisp/dhis/rules/models/RuleActionAssign.kt
package org.hisp.dhis.rules.models

data class RuleActionAssign(
    val field: String,
    val content: String,
    val attributeType: AttributeType,
    val data: String
) : RuleActionAttribute {
    companion object {
        fun create(
            content: String?,
            data: String, field: String?, attributeType: AttributeType?
        ): RuleActionAssign {
            require(!(content == null && field == null)) {
                "Either content or field " +
                        "parameters must be not null."
            }
            return RuleActionAssign(
                field ?: "", content ?: "",attributeType ?: AttributeType.UNKNOWN , data

            )
        }

        fun create(content: String?, data: String, field: String?): RuleActionAssign {
            return create(content, data, field, AttributeType.UNKNOWN)
        }
    }

    override fun attributeType(): AttributeType {
        return attributeType
    }

    override fun data(): String {
        return data
    }
}
